### PR TITLE
Issue #3601: Added examples for MethodParamPad xdocs

### PR DESCRIPTION
**Pull request by SergeyDzyuba**
Issue: #3601

### CODE REVIEW

**Comment by MEZk** [comment]
@SergeyDzyuba 
Please, use Present tense in commit messages. For example: ```Issue #####: Split input test```` . The name of the commit describes what the commit does (add, fix, edit, split, etc). Rather than writing messages that say what a committer has done. It's better to consider these messages as the instructions for what is going to be done after the commit is applied on the repo. Thanks.

**Comment by romani** [comment]
Please rebase other our master and make sure that you send us only your changes

**Comment by MEZk** [comment]
@SergeyDzyuba 
Use ```git rebase -i HEAD~N``` (where N is a number of commits to analyze) to amend/rebase/squash your commits.

**Comment by MEZk** [comment]
@SergeyDzyuba 
AppVeyor fails due to:
```
[00:01:15] execute:
[00:01:15]      [echo] Checkstyle started: 14/03/2017 09:35:31 PM
[00:01:16] [checkstyle] Running Checkstyle 7.7-SNAPSHOT on 930 files
[00:01:52] [checkstyle] [ERROR] C:\projects\checkstyle\src\xdocs\config_whitespace.xml:772: Unnecessary consecutive lines [RegexpMultiline]
[00:01:52]      [echo] Checkstyle finished: 14/03/2017 09:36:09 PM
```

**Comment by SergeyDzyuba** [comment]
@MEZk i will try again to use mvn clean verify

**Comment by romani** [comment]
content of example should be rewritten:
You need: 
```
To configure the check by default
CONFIG
<Example of code with violations and without with trailing comments to show a place of violation> 

To configure the check to require NO space after the identifier but allow wrapping to newline
<CONFIG>
<EXample with trailing comments on where violation will and not>
```

File: src/xdocs/config_whitespace.xml [hunk 1]
```diff
@@ -719,29 +719,59 @@ sort(list, Comparable::&lt;String&gt;compareTo);             // Method reference
...
+class Foo
+{
+    public void foo
+    () {}
+}
+            </source>
+        </subsection>
 
-      <subsection name="Example of Usage">
+        <subsection name="Example of Usage">
```
Comment (rnveach) [inline src/xdocs/config_whitespace.xml]: Unnecessary change. Please revert.

File: src/xdocs/config_whitespace.xml [hunk 2]
```diff
@@ -719,29 +719,59 @@ sort(list, Comparable::&lt;String&gt;compareTo);             // Method reference
...
-        </p>
-        <source>
-&lt;module name=&quot;MethodParamPad&quot;&gt;
-    &lt;property name=&quot;tokens&quot; value=&quot;METHOD_DEF&quot;/&gt;
-    &lt;property name=&quot;option&quot; value=&quot;space&quot;/&gt;
-    &lt;property name=&quot;allowLineBreaks&quot; value=&quot;true&quot;/&gt;
-&lt;/module&gt;
-        </source>
-      </subsection>
+        <subsection name="Examples">
```
Comment (rnveach) [inline src/xdocs/config_whitespace.xml]: Can we please try to use original indentation so we don't have this change?

**Comment by SergeyDzyuba** [comment]
@rnveach i have fixed intendation
@romani i wrote new examples. are they better now?

**Comment by sabaka** [comment]
@SergeyDzyuba , have you pushed your changes?
I still see indentation problem, requested above.

**Comment by SergeyDzyuba** [comment]
@sabaka yep, my bad. i've fixed that

**Comment by romani** [comment]
please rebase on our latest code , I just merged a fix for wercker CI to our master.

**Comment by rnveach** [comment]
@SergeyDzyuba ping

**Comment by SergeyDzyuba** [comment]
@rnveach got it

**Comment by romani** [comment]
Please reformat documentation to make it like: config and examples of it validation.

Please update code samples and put word "violation, ..." In front of trailing comment content. Use  care about fact valid or violation more than explanation why.

**Comment by rnveach** [comment]
@SergeyDzyuba ping
Since we are out of proposal deadline, do you think you can finish this?

**Comment by rnveach** [comment]
@SergeyDzyuba If there is no activity by May 1st, PR will be closed as incomplete and issue will be open to others.

**Comment by romani** [comment]
@SergeyDzyuba , ping

**Comment by romani** [comment]
PRs is closed as we lost connection with student, any contributor could continue this work to make it done.
